ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java

```diff
@@ -530,11 +530,19 @@ private static FlatTrace.Context handleCallDataLoad(
 
   private static boolean hasRevertInSubCall(
       final TransactionTrace transactionTrace, final TraceFrame callFrame) {
-    return transactionTrace.getTraceFrames().stream()
-        .filter(traceFrame -> !traceFrame.equals(callFrame))
-        .takeWhile(traceFrame -> !traceFrame.getOpcode().equals("RETURN"))
-        .filter(traceFrame -> traceFrame.getOpcode().equals("REVERT"))
-        .anyMatch(traceFrame -> traceFrame.getDepth() == callFrame.getDepth());
+    for (int i = 0; i < transactionTrace.getTraceFrames().size(); i++) {
+      if (i + 1 < transactionTrace.getTraceFrames().size()) {
+        final TraceFrame next = transactionTrace.getTraceFrames().get(i + 1);
+        if (next.getDepth() == callFrame.getDepth()) {
+          if (next.getOpcodeNumber() == 0xFD) { // REVERT opCode
```

Comment:
Consider adding an OPCODE constant to RevertOperation and ReturnOperation so we can reduce magic constants in the code.

Reply:
Thanks for the suggestion, I was thinking actually of an Enum OpCode to store all the opcodes and use only the Enum in the EVM and outside, like Opcode.REVERT.getOpcodeNumber() for 0xFD
public enum Opcode {
    STOP(0x00, "STOP"),
    ADD(0x01, "ADD"),
    MUL(0x02, "MUL"),
    ...
}
In terms of performance, I guess JIT will replace by the opCodeNumber or the opcode name at runtime

Reply:
@shemnon I've added the OpCode enum within the EVM and would like to use it across all EVM opcode operations. An implementation example has been provided for ADDMOD. Does this approach work for you ?

Reply:
This is not what I meant. Such an OpCode enum breaks exensibility downstream users use.  Please revert.

Reply:
What I intended was something like this, where a constant is added to the Operation classes, not an entirely different enum.

https://github.com/hyperledger/besu/blob/c89e4db8c345f0330774ca27ccfccff35bdbe20c/evm/src/main/java/org/hyperledger/besu/evm/operation/BlobHashOperation.java#L37

Reply:
I actually understood your suggestion but wanted to suggest an implementation that would work for similar use cases in the future, but it looks it wasn't a good idea. I'm going to revert the last changes.